### ISSUE TITLE
--focus not working with "When" nodes

### ISSUE THREAD

**Issue by Cirivas** [onsi/ginkgo#1226]
Hi,

I have the following test:

```
func TestErrors(t *testing.T) {
	RegisterFailHandler(Fail)
	RunSpecs(t, "Test Suite")
}

var _ = Describe("Test case", func() {
	When("function is called", func() {
		It("should return correctly", func() {
			Expect(1).To(Equal(1))
		})
	})
})
```

Using `gingko run -r -focus "Test case function is called should return correctly"` returns the following:

```
Will run 0 of 1 specs
S

Ran 0 of 1 Specs in 0.000 seconds
SUCCESS! -- 0 Passed | 0 Failed | 0 Pending | 1 Skipped
PASS

Ginkgo ran 1 suite in 1.912395541s
Test Suite Passed
```

But changing `When` node to a `Context` node runs the test correctly:

``` 
Will run 1 of 1 specs
•

Ran 1 of 1 Specs in 0.001 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 0 Skipped
PASS

Ginkgo ran 1 suite in 2.366964375s
Test Suite Passed
```

Using a `When` node completely ruins the output. For example:

```
var _ = Describe("Test case", func() {
	Context("function is called", func() {
		When("anther node", func() {
			It("should return correctly", func() {
				Expect(1).To(Equal(1))
			})
		})

	})
})
```

Running `ginkgo -r -focus "Test case function is called"` runs test cases correctly, but `ginkgo -r -focus "Test case function is called"` or anything thats inside the `When` node fails. Per documentation, `When` and `Context` nodes should be functionally equivalent aliases, but they are changing the behaviour of --focus.

I'm currently using Ginkgo Version 2.11.0.

**Comment by onsi** [comment]
hey @Cirivas when I first shipped the beta/release candidate for 2.0 I had `When` behave exactly like `Context`.  But I got feedback asking for me to revert the behavior to how 1.0 works - namely - when using `When` the word `when` is injected into the concatenated spec string and you'll need to add `when` to your focus filter.

So, the documentation is wrong - and I should update it to reflect that `When` is not a strict alias of `Context` but also has this behavior of adding `when` to the spec text.

While `--focus` is a fully supported way to pick a specific test or set of tests to run I do recommend the use of `Label`s and `--label-filter` where possible to get better control of spec organization and filtering.
